### PR TITLE
Director separation

### PR DESCRIPTION
**Pull request by martijnjanssen**
Changed functionality of the buttons to adapt to the correct behavior.

### CODE REVIEW

File: src/main/java/gui/centerarea/DirectorShotBlock.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        this.getTimetableBlock().setPrefHeight(grid.getNumberOfVerticalGrids()
+                * grid.getVerticalElementSize());
+        grid.addDirectorShotBlock(this);
+    }
+
+    /**
+     * Recompute the position in the grid and repaint with new values.
+     */
+    @Override
+    public void recompute() {
```
Comment (geenen124) [inline src/main/java/gui/centerarea/DirectorShotBlock.java]: Isn't this unnecessary if all you're doing is calling the super method?

File: src/main/java/gui/events/CameraShotCreationEvent.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package gui.events;
+
+import lombok.Getter;
+
+import java.util.List;
+
+/**
+ * Class that represents a director shot creation event.
```
Comment (geenen124) [inline src/main/java/gui/events/CameraShotCreationEvent.java]: Shouldn't this say camera shot?

File: src/main/java/gui/events/CameraShotCreationEvent.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * @author martijn
+ */
+public class CameraShotCreationEvent extends ModalEvent {
+
+    @Getter
+    private String shotName;
+    @Getter
+    private String shotDescription;
+    @Getter
+    private List<Integer> camerasInShot;
```
Comment (geenen124) [inline src/main/java/gui/events/CameraShotCreationEvent.java:18]: Isn't this for a single camera shot?

File: src/main/java/control/ToolViewController.java [hunk 4]
```diff
@@ -93,26 +97,41 @@ private void showBlockCreationWindow(MouseEvent event) {
...
-                                             event.getShotStart(), event.getShotEnd());
-
-        TimelineController timelineController = this.controllerManager.getTimelineControl();
+    private void createCameraShot(CameraShotCreationEvent event) {
+        TimelineController tlineControl = this.controllerManager.getTimelineControl();
         event.getCamerasInShot().forEach(camInd -> {
-                timelineController.addCameraShot(camInd, event.getShotName(),
-                        event.getShotDescription(), event.getShotStart(),
-                        event.getShotEnd());
+                tlineControl.addCameraShot(camInd,
```
Comment (geenen124) [inline src/main/java/control/ToolViewController.java:119]: Shouldn't this create a single camera shot?

Comment (martijnjanssen) [inline src/main/java/control/ToolViewController.java:119]: This creates the camera shots in the checked timelines. This is not the create method for the DirectorShot.

File: src/main/java/gui/events/DirectorShotBlockUpdatedEvent.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import javafx.event.Event;
+import javafx.event.EventType;
+
+/**
+ * Created by Bart.
+ */
+public class DirectorShotBlockUpdatedEvent extends ShotblockUpdatedEvent {
+
+    /**
+     * The eventtype.
```
Comment (Markvanderuit) [inline src/main/java/gui/events/DirectorShotBlockUpdatedEvent.java]: Isn't javadoc a bit overkill here?

File: src/main/java/gui/events/CameraShotCreationEvent.java (hunk 3 above)
Comment (martijnjanssen) [inline src/main/java/gui/events/CameraShotCreationEvent.java:18]: This also refers to the shots in the timeline, not the DirectorShot. The javadoc is incorrect. I will change that.

File: src/main/java/gui/events/CameraShotCreationEvent.java (hunk 2 above)
Comment (martijnjanssen) [inline src/main/java/gui/events/CameraShotCreationEvent.java]: Yes.

File: src/main/java/gui/root/RootCenterArea.java [hunk 6]
```diff
@@ -31,6 +31,8 @@
     @Getter
     private int counterWidth = 10;
     @Getter
+    private int directorTimelineWidth = 200;
```
Comment (geenen124) [inline src/main/java/gui/root/RootCenterArea.java]: Make this a static final variable or move initialization to the constructor

**Comment by geenen124** [comment]
Looks good!

**Pull request merged**